### ISSUE TITLE
Preserve TM:PE settings on segment update

### ISSUE THREAD

**Issue by HypoSim** [CitiesSkylinesMods/TMPE#1013]
Currently, all TM:PE settings are reset when a segment is upgraded, even if the same network is used.

**Comment by krzychu124** [comment]
Yeah, this needs investigation, probably it will need another chunk of harmony integration in segment patch before it will be possible
@kianzarrin have you tried solve similar problem in any of your mods?

**Comment by kianzarrin** [comment]
@krzychu124 I have alread some code that works for node controller.

In general I think it is a drag that every mod has to do all this. this is a complicated process. I think It would be a good idea to write a library that allows each mod to implement INetowrkPersistancy . something like MoveITIntegration but better

```
Copy
Paste
Encode
Decode
Merge
Split
```

- the methods above can be called as approprate when a segment is split when the user attaches a new segment.... when a node is moved ... when node is inserted/removed by elektrix road toos or when MoveIT copy pastes stuff.
- They can also be called if to load/store network data from intersection editor.
- maybe even allow for undo/redo .

Backward compatiblity would be an issue for mods that already have implemented these things.
